### ISSUE TITLE
Error "unimplemented type 'list' in 'orderVector1'" in Rstudio

### ISSUE THREAD

**Issue by JacekPardyak** [rstudio/reticulate#1438]
Dear team
I get this error in Rstudio when trying to order character column containing NULL in dataframe returned by Reticulate. Below are two examples:
```
def build_py_df():
  np.random.seed(6520)
  random_df = pd.DataFrame(
    {'group': np.random.choice(data_tools, 500),
    'int': np.random.randint(1, 10, 500),
    'num': np.random.randn(500),
    'char': [''.join(np.random.choice(list(alpha), 3)) for _ in range(500)],
    'bool': np.random.choice([True, False], 500),
    'date': np.random.choice(pd.date_range('2000-01-01', '2019-05-31'), 500),
    'new': np.random.choice([1, np.NaN], 500)
    })
  return random_df
```
and

```
def build_py_df_fix():
  np.random.seed(6520)
  random_df = pd.DataFrame(
    {'group': np.random.choice(data_tools, 500),
    'int': np.random.randint(1, 10, 500),
    'num': np.random.randn(500),
    'char': [''.join(np.random.choice(list(alpha), 3)) for _ in range(500)],
    'bool': np.random.choice([True, False], 500),
    'date': np.random.choice(pd.date_range('2000-01-01', '2019-05-31'), 500),
    'new': np.random.choice([1, np.NaN], 500)
    })
  tmp = random_df.select_dtypes(include='object')
  tmp.fillna("NULL", inplace = True)
  random_df = tmp.join(random_df.select_dtypes(exclude='object'))
  return random_df
```
Ordering values in column `group` of dataframe produced with `df = build_py_df()` call will produce error:

<img width="323" alt="Screenshot 2023-08-10 114511" src="https://github.com/rstudio/reticulate/assets/15980055/58307e0f-be69-42d5-8eaf-c5cd18bb320e">

Ordering numeric column containing `NULL` works perfectly.
This is my quick fix : `df_fix = build_py_df_fix()` to get table where ordering works fine.
Wish you nice day 👍

**Comment by dfalbel** [comment]
Hi @JacekPardyak , 

Thanks for reporting. The issue can be separated in 2 problems:

1. You expected that string columns containing `None` in Python were converted into R character vectors (where `NULL` would be represented by `NA`s, but instead it returned a list column - you can see it if you do `str(reticulate::py$build_py_df())`. THis behavior has been fixed in #1428. You can install the dev version of reticulate or wait for the next CRAN release.

2. There's an issue in the RStudio IDE Viewer, triggered by trying to order a column that is actually a list column. This problem can be independently of reticulate with: 

```
df <- data.frame(x = 1:3)
df$y <- list("a", "b", "c")
df
View(df)
```

**Comment by JacekPardyak** [comment]
Thank you @dfalbel 👍 
Installing the dev version solved the problem.
